near-sdk/src/json_types/account.rs

```diff
@@ -0,0 +1,84 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::Serialize;
+use std::convert::{TryFrom, TryInto};
+
+use crate::env::is_valid_account_id;
+use crate::AccountId;
+
+/// Helper class to validate account ID during serialization and deserializiation
+#[derive(
+    Debug, Clone, PartialEq, PartialOrd, Ord, Eq, BorshDeserialize, BorshSerialize, Serialize,
+)]
+pub struct ValidAccountId(pub AccountId);
+
+impl ValidAccountId {
+    pub fn is_valid(&self) -> bool {
```

Comment:
this doesn't make sense. So something of type `ValidAccountId` can still be invalid?

Reply:
I guess I can make it private. In theory I can make inner field private, which will hide the constructor, so you can only construct it from the `TryFrom`.

Reply:
>  In theory I can make inner field private, which will hide the constructor, so you can only construct it from the TryFrom.

Sounds good.